libs/product/state/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -99,6 +103,27 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
     }),
   );
 
+  const selectCompositeProductDiscountAmount = createSelector(
+    selectProductEntities,
+    selectCompositeProductAppliedOptionsEntities,
+    (products, appliedOptionsEntities, props) => {
+      const product = selectProduct.projector(products, { id: props.id });
+      if(product.type !== DaffProductTypeEnum.Composite) {
+        return undefined;
+      }
+
+      const appliedConfigurationItem = appliedOptionsEntities[product.id].items;
+      if((<DaffCompositeProduct>product).items.filter(item => item.required && appliedConfigurationItem[item.id].value === null).length > 0) {
```

Comment:
It would be more performant to implement this with `indexOf` since you're only looking for a single element that matches this criteria. As soon as you find it, you know that you can exit and return, you don't need to keep checking the rest of the list.

Reply:
I think `indexOf` won't work, because I'd need to pass it an equivalent object rather than being able to define a custom condition. Am I crazy?

Reply:
you are not, I meant to say `findIndex`

Reply:
changed

Reply:
ah sorry...your original implementation here is actually the only one that works on IE11. I though `findIndex` had IE support...

Reply:
np, reverted